sigar_linux.go

```diff
@@ -88,11 +89,25 @@ func (self *ProcFDUsage) Get(pid int) error {
 	if err != nil {
 		return err
 	}
-	fds, err := ioutil.ReadDir(procFileName(pid, "fd"))
+
+	//no point to get FD usage if we are not "root" , this will trigger audit if DISA_STIG has been applied
+	//DISA_STIG V-72127 - All users of the openat command must be audited
+	//on busy systems this will flood audit daemon specially if its continuously running (metricbeat) without root permission
+
+	user, err := user.Current()
 	if err != nil {
 		return err
 	}
-	self.Open = uint64(len(fds))
+	if user.Uid == "0" {
```

Comment:
I think there is a problem in that it is possible for non-root users to access /proc/pid/fd information via capabilities (https://stackoverflow.com/questions/41631927/read-proc-pid-fd-fd-without-full-root-access).

Maybe the caller should stop repeating the calls if it gets an access denied error.

Reply:
Well I think its not a good idea in safety/mission critical systems to give CAP_DAC_READ_SEARCH access to a user/process which are not allowed to run under root privileges since CAP_DAC_READ_SEARCH will also allow to access other files as well and not only FD.

The caller will get access denied however still the event will be logged under audit.